### PR TITLE
Edit function

### LINKED ISSUE #4: Edit encounter
## Value statement
As a Game Master
I need a preview of already submitted entries with a delete and edit button
so that I can keep track of which entries I have added and to edit and delete them when needed

## Description (Text, Scribble, Wireframe, Design)
image

## Acceptance criteria
- [x] When the first entry is submitted the "Show created entries" button is displayed
- [x] Clicking the "Show created entries" button toggles the display of an entry preview
- [x] Every entry preview has a delete and edit button
- [x] Clicking the delete button deletes the entry
- [x] Clicking the edit button deletes the entry but enters the entry's values into the submit form to be edited and resubmitted

## Tasks
- [x] Create feature branch
- [x] Create EnemyPreview component in storybook
- [x] Create PlayerPreview component in storybook
- [x] Create EditButton component in storybook
- [x] Create EditPage component in storybook
- [x] Add edit and delete logic
- [x] Write component tests
- [x] Add components to app
- [x] Add missing styles
- [x] Write integration tests w/ Cypress

## Size
large

### PR DESCRIPTION
**Pull request by ninasander**
Please review this pull request linked to this story:
#4 

As a Game Master
I need a preview of already submitted entries with a delete and edit button
so that I can keep track of which entries I have added and to edit and delete them when needed

This pull request turned out larger than anticipated, so an extra big thank you to anyone who reviews :)

### CODE REVIEW

File: src/components/CreatePage/PlayerPreview.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  handleEdit: PropTypes.func.isRequired,
+}
+
+export default function PlayerPreview({
+  handleDelete,
+  handleEdit,
+  creatureEntry,
+}) {
+  return (
+    <div>
```
Comment (constantingoeldel) [inline src/components/CreatePage/PlayerPreview.js]: If the `div` is solely used to group these together, you might want to use a fragment instead as it does not show up in the final build

File: src/components/lib/localStorage.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export function loadLocally(key) {
+  try {
+    const serializedData = localStorage.getItem(key)
+    if (serializedData) {
```
Comment (constantingoeldel) [inline src/components/lib/localStorage.js:4]: This should be sufficient
```suggestion
    serializedData && JSON.parse(serializedData)
```

File: src/components/lib/localStorage.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    }
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+export function saveLocally(key, value) {
+  try {
+    const serializedData = JSON.stringify(value)
+    localStorage.setItem(key, serializedData)
```
Comment (constantingoeldel) [inline src/components/lib/localStorage.js:15]: This way you can safe a line. I haven't implemented local Storage yet, so I'm curious: Is the async approach necessary? 
```suggestion
    localStorage.setItem(key, JSON.stringify(value))
```

File: src/components/hooks/usePlayerPreviewHeight.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { useLayoutEffect, useState, useRef } from 'react'
+
```
Comment (constantingoeldel) [inline src/components/hooks/usePlayerPreviewHeight.js:2]: I have no idea what you are doing in this file, can you explain it to me?  🤷‍♂️

File: src/components/CreatePage/CreatePlayer.js [hunk 5]
```diff
@@ -1,18 +1,39 @@
...
+  addCreatureEntry,
+  editCreatureEntry,
+  editCreature,
+}) {
+  const { register, handleSubmit, errors, setValue } = useForm()
+
+  const onSubmit = (creatureEntry, event) => {
     event.target.reset()
-    addCreatureEntry(playerEntry)
+    if (editCreature) {
```
Comment (constantingoeldel) [inline src/components/CreatePage/CreatePlayer.js:16]: A ternary might simplify this a bit

File: src/components/CreatePage/CreatePage.js [hunk 6]
```diff
@@ -1,25 +1,99 @@
...
+            isPlayerPreviewVisible
+              ? 'Hide Created Players ▲'
+              : 'Show Created Players ▼'
+          }
+          onClick={togglePlayerPreview}
+        />
+      ) : null}
+      <PreviewContainerStyled {...bindPlayer} style={playerPreviewStyle}>
+        {creatureEntries.map((creatureEntry) =>
+          creatureEntry.playerName ? (
```
Comment (constantingoeldel) [inline src/components/CreatePage/CreatePage.js]:
```suggestion
          creatureEntry.playerName && ...
```

**Review by constantingoeldel — APPROVED** [review]
Impressive progress! I left some comments concerning minor code improvements but can't really say anything about the general structure of your app this quickly so I might have missed some things

File: src/components/lib/localStorage.js (hunk 3 above)
Comment (ninasander) [reply src/components/lib/localStorage.js:15]: Good question, I haven't tried it without the async, but now I'll test it out to see if it's really necessary. Thanks!

File: src/components/hooks/usePlayerPreviewHeight.js (hunk 4 above)
Comment (ninasander) [reply src/components/hooks/usePlayerPreviewHeight.js:2]: Both this and the useEnemyPreviewHeight hook are helper functions to calculate the individual height of my preview components, so the show/hide animation works smoothly. Did that help? If not just ask me on Slack :)

**Pull request merged**